.github/workflows/publish.yml

```diff
@@ -0,0 +1,43 @@
+name: Publish Docker image
+on:
+# Will only run when release is published. 
+  push:
+  release:
```

Comment:
Does this mean it's tied to the github release feature? In other words, does this mean we need to publicly publish a release _before_ it is built?

Reply:
yes. But we can change it to release creates which will get trigger even if we draft the release. But build, tests and other are running on create release anyways so this is just a publishing artifacts once we publish release.

Reply:
In my mind, the ideal flow would be creating a release would depend on the output of this. Basically, we don't want to broadcast a release if it's not actually available for download it yet (also, if anything goes wrong with publishing, that gives us space to fix it). If the closest we can get to that is triggering on draft, that could work.